### PR TITLE
Add Amazon + Instagram examples

### PR DESCRIPTION
**Pull request by jeffreyyun**
Add Amazon based on https://github.com/anon-dot-com/examples/blob/r11bump/run-typescript/index.ts#L59C1-L91C3

## Testing
**How to test**
First, make sure your Amazon account is linked:

```
cd run-typescript
```
Then in `actions/config.ts`, change the `DEFAULT_APP` to `"amazon"`.

Then run the playwright script
```
yarn run dev
```

**Cart screenshot**
image

**Log output**
```
 ~/anon/examples/run-typescript   anon1454  yarn run dev               
yarn run v1.22.22
$ npx tsx index.ts
Starting script execution...
Environment variables loaded.
Configuration set:
APP_USER_ID: Set
API_KEY: Set
ANON_ENV: sandbox
APP: amazon
Creating Anon client...
Anon client created.
Starting main function...
Requesting amazon session for app user id "41da5feb-4a06-401f-ad2b-475d3a189859"...
Setting up Anon browser with context...
Anon browser setup complete.
Getting first page from browser context...
Page acquired.
Executing runtime script...
Starting amazonAddAirpodsToCart function...
Step 0: Navigating to Amazon homepage...
Successfully navigated to Amazon.
Screenshot taken: screenshot-amazon-0-amazon-homepage-loaded-1720557409087.png
Step 1: Searching for AirPods...
Search submitted for AirPods.
Screenshot taken: screenshot-amazon-1-search-submitted-1720557413676.png
Step 2: Waiting for search results...
Search results loaded.
Screenshot taken: screenshot-amazon-2-search-results-loaded-1720557415010.png
Step 3: Navigating to AirPods product page...
Navigated to AirPods product page.
Screenshot taken: screenshot-amazon-3-product-page-loaded-1720557422746.png
Step 4: Waiting for 'Add to Cart' button...
'Add to Cart' button is visible and enabled.
Screenshot taken: screenshot-amazon-4-add-to-cart-visible-1720557428015.png
Step 5: Clicking 'Add to Cart' button...
'Add to Cart' button clicked successfully.
Screenshot taken: screenshot-amazon-5-add-to-cart-clicked-1720557432883.png
Step 6: Handling insurance option...
Insurance option handled.
Screenshot taken: screenshot-amazon-6-insurance-option-handled-1720557438449.png
Step 7: Navigating to Cart...
Navigated to Cart.
Screenshot taken: screenshot-amazon-7-cart-navigated-1720557443384.png
Step 8: Waiting for Checkout Button...
Checkout button is visible.
Screenshot taken: screenshot-amazon-8-checkout-button-visible-1720557445602.png
Amazon Airpods addition to cart process completed.
Runtime script execution completed.
Script execution finished successfully.
Script execution completed.
✨  Done in 53.57s.
```

### CODE REVIEW

**Comment by jeffreyyun** [comment]
Attempted Instagram `sendMessageToSelf` haven't gotten it working

Here for reference
```
// TODO: get this working
export const instagramSendMessageToSelf = async (page: Page) => {
//   // Navigate to Instagram feed (assuming you're already logged in)
//   await page.goto('https://www.instagram.com/');

//   // Click on the profile icon to go to your profile page
//   // Note: This selector might need to be adjusted based on Instagram's current layout
//   await page.click('svg[aria-label="Profile"]');
//   await page.waitForNavigation();

//   // Extract username from the profile page
//   // Note: This selector might need to be adjusted based on Instagram's current layout
//   const usernameElement = await page.$('.username');
//   const myUsername = await page.evaluate(el => el?.textContent, usernameElement);

//   console.log(`Extracted Username: ${myUsername}`);
  const myUsername = "adminitestero";

  // Open the Direct Messages page
  await page.click('text=Direct');
  await page.waitForNavigation();

  // Click on the pencil icon to start a new message
  await page.click('text=New Message');
  await page.waitForSelector('input[placeholder="Search..."]');

  // Type your extracted username to send a message to yourself
  await page.fill('input[placeholder="Search..."]', myUsername!);
  await page.press('input[placeholder="Search..."]', 'Enter');
  await page.waitForSelector('div[role="dialog"] button[type="submit"]');

  // Click on the name that appears (your profile) to open the chat
  await page.click('div[role="dialog"] button[type="submit"]');
  await page.waitForNavigation();

  // Type your message
  const message = "Hello! This is an automated message.";
  await page.fill('textarea[placeholder="Message..."]', message);
  await page.press('textarea[placeholder="Message..."]', 'Enter');

  // Wait a bit before closing the browser
  await page.waitForTimeout(5000);
};
```

File: run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
+import { Page } from "playwright";
+import { retryWithBackoff, takeScreenshot } from "../browserHelpers";
+import { NETWORK_TIMEOUT } from "../config";
+
+const AMAZON_URL = "https://amazon.com";
+
+export const amazonAddAirpodsToCart = async (page: Page) => {
```
Comment (jeffreyyun) [inline run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts:7]: Adapted from https://github.com/anon-dot-com/examples/blob/r11bump/run-typescript/index.ts#L59-L112

File: run-typescript/actions/config.ts [hunk 2]
```diff
@@ -1,15 +1,29 @@
 import { Page } from "playwright";
 import { linkedInCreatePost } from "./linkedin/linkedInCreatePost";
+import { instagramNavigateToMessages } from "./instagram/instagramMessage";
+import { amazonAddAirpodsToCart } from './amazon/amazonAddHeadphonesToCart'
+
+export const DEFAULT_APP: AppName = "linkedin";
+export const NETWORK_TIMEOUT = 10000; // 10 seconds
+export const MAX_RETRIES = 5;
```
Comment (jeffreyyun) [inline run-typescript/actions/config.ts:8]: moved these configs to top for easier changing

File: run-typescript/actions/instagram/instagramMessage.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+  console.log("Step 1: Navigating to Instagram Messages...");
+  await retryWithBackoff(async () => {
+    await page.goto(INSTAGRAM_MESSAGES_URL, { timeout: NETWORK_TIMEOUT });
+    console.log("Successfully navigated to Instagram Messages.");
+  });
+  await takeScreenshot(page, "instagram", "1-instagram-messages");
+
+  console.log("Step 2: Waiting for page to load...");
+  await waitForNetworkIdle(page);
+  await takeScreenshot(page, "instagram", "2-after-page-load");
```
Comment (jeffreyyun) [inline run-typescript/actions/instagram/instagramMessage.ts:19]: simply navigating to Messages, waiting for whole thing to load, and having those screenshots yields a great POC for our using the IG session.

my vision was sending a message to yourself, for that extra "wow", but couldn't get it working within my timebox -- https://github.com/anon-dot-com/examples/pull/17#issuecomment-2218380376

File: run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+        console.log("Navigated to Cart.");
+        await takeScreenshot(page, "amazon", "7-cart-navigated");
+    });
+
+    // Step 8: Wait for Checkout Button
+    console.log("Step 8: Waiting for Checkout Button...");
+    await retryWithBackoff(async () => {
+        await page.waitForSelector('input[name="proceedToRetailCheckout"]', { state: "visible" });
+        console.log("Checkout button is visible.");
+        await takeScreenshot(page, "amazon", "8-checkout-button-visible");
```
Comment (samgreen) [inline run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts:86]: This is super easy to read / follow!

File: run-typescript/actions/config.ts [hunk 5]
```diff
@@ -1,15 +1,29 @@
...
 
 export interface AppConfig {
   url: string;
   action: (page: Page) => Promise<void>;
 }
 
 export const APP_CONFIG: { [key: string]: AppConfig } = {
+  amazon: {
+    url: "https://www.amazon.com",
```
[the remark this answers is not in the document]
Comment (jeffreyyun) [reply run-typescript/actions/config.ts:17]: hmm yeah the `AMAZON_URL` is just there for https://github.com/anon-dot-com/examples/pull/17/files#diff-78822724e9e62eac13f0660052c22d093305950337b4f733e6ded7a14a64fe15R10 -- which is not strictly needed but demonstrative (if they run into any problems, at least Step 0 screenshot holds)

File: run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+      await takeScreenshot(page, "amazon", "1-search-submitted");
+    });
+
+    // Step 2: Wait for navigation after the search
+    console.log("Step 2: Waiting for search results...");
+    await page.waitForSelector(".s-main-slot.s-result-list");
+    console.log("Search results loaded.");
+    await takeScreenshot(page, "amazon", "2-search-results-loaded");
+
+    // TODO: automagically select first product
```
Comment (grant) [inline run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts:34]: Remove this TODO unless the sample completely fails without this.

File: run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts [hunk 7]
```diff
@@ -0,0 +1,90 @@
...
+import { retryWithBackoff, takeScreenshot } from "../browserHelpers";
+import { NETWORK_TIMEOUT } from "../config";
+
+const AMAZON_URL = "https://amazon.com";
+
+export const amazonAddAirpodsToCart = async (page: Page) => {
+    console.log("Starting amazonAddAirpodsToCart function...");
+
+    // Step 0: Navigate to Amazon
+    console.log("Step 0: Navigating to Amazon homepage...");
```
Comment (grant) [inline run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts:11]: I would remove each of the step comments and inline the details inside the `console.log` statement. 

If the strings differ, then choose the more detailed one.

**Review by grant — APPROVED** [review]
Looks good, thanks! Please add a detailed description for the PR including instructions on how to use/test the code, and the console.log result output after running this code.

File: run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts (hunk 7 above)
Comment (jeffreyyun) [reply run-typescript/actions/amazon/amazonAddHeadphonesToCart.ts:11]: Darn I liked the syntax highlighting + how the comment goes more into depth about what the code does

Sure, removing...

**Pull request merged**